### PR TITLE
[DependencyInjection] Avoid generating call_user_func in more cases

### PR DESCRIPTION
**Pull request by realityking**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

This is a follow up to #9807 to add the PHP 5.4 only coded I excluded when I split it off #9432.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -542,6 +542,10 @@ private function addServiceConfigurator($id, $definition, $variableName = 'insta
                 return sprintf("        %s::%s(\$%s);\n", $this->dumpLiteralClass($class), $callable[1], $variableName);
             }
 
+            if (strpos($class, "new") === 0) {
```
Comment (sstok) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Can be single quotes.

**Comment by realityking** [comment]
Not sure what do with the fabbot.io failures as the offending code is in the dumped containers.

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 2]
```diff
@@ -542,6 +542,10 @@ private function addServiceConfigurator($id, $definition, $variableName = 'insta
                 return sprintf("        %s::%s(\$%s);\n", $this->dumpLiteralClass($class), $callable[1], $variableName);
             }
 
+            if (strpos($class, 'new') === 0) {
```
Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Shouldn't this check for `new ` (note the trailing space)?

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: I agree, should be `0 === strpos($class, 'new ')`

**Comment by nicolas-grekas** [comment]
:+1: after @xabbuh's comment is fixed (fabbot failures should be ignored for failures).

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 3]
```diff
@@ -723,6 +727,10 @@ private function addNewInstance(Definition $definition, $return, $instantiation,
                     return sprintf("        $return{$instantiation}%s::%s(%s);\n", $this->dumpLiteralClass($class), $callable[1], $arguments ? implode(', ', $arguments) : '');
                 }
 
+                if (strpos($class, 'new') === 0) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: `0 === strpos($class, 'new ')`

**Comment by nicolas-grekas** [comment]
ping @realityking

**Comment by javiereguiluz** [comment]
@realityking will you have some time to finish this nice pull request adding the minor changes asked by @nicolas-grekas? If not, we can make those minor changes ourselves. Thank you!

**Comment by realityking** [comment]
Sorry, I was distracted and then forgot all about this. Comments addressed, hope Travis runs succeeds this time.

**Comment by realityking** [comment]
Remaning failures are in the generated code.

**Comment by nicolas-grekas** [comment]
Thank you @realityking.

**Pull request closed without merging**